python/cudf/cudf/tests/test_parquet.py

```diff
@@ -3243,3 +3244,106 @@ def test_parquet_reader_zstd_huff_tables(datadir):
     expected = pa.parquet.read_table(fname).to_pandas()
     actual = cudf.read_parquet(fname)
     assert_eq(actual, expected)
+
+
+def test_parquet_reader_roundtrip_with_arrow_schema():
+    # Ensure that the nested types are faithfully being roundtripped
+    # across Parquet with arrow schema which is used to faithfully
+    # round trip duration types (timedelta64) across Parquet read and write.
+    pdf = pd.DataFrame(
+        {
+            "s": pd.Series([1234, 3456, 32442], dtype="timedelta64[s]"),
+            "ms": pd.Series([1234, 3456, 32442], dtype="timedelta64[ms]"),
+            "us": pd.Series([1234, 3456, 32442], dtype="timedelta64[us]"),
+            "ns": pd.Series([1234, 3456, 32442], dtype="timedelta64[ns]"),
+            "duration_list": list(
+                [
+                    [
+                        datetime.timedelta(minutes=7, seconds=4),
+                        datetime.timedelta(minutes=7),
+                    ],
+                    [
+                        datetime.timedelta(minutes=7, seconds=4),
+                        datetime.timedelta(minutes=7),
+                    ],
+                    [
+                        datetime.timedelta(minutes=7, seconds=4),
+                        datetime.timedelta(minutes=7),
+                    ],
+                ]
+            ),
+            "int64": pd.Series([1234, 123, 4123], dtype="int64"),
+            "list": list([[1, 2], [1, 2], [1, 2]]),
+            "datetime": pd.Series([1234, 123, 4123], dtype="datetime64[ms]"),
+            "map": pd.Series(["cat", "dog", "lion"]).map(
+                {"cat": "kitten", "dog": "puppy", "lion": "cub"}
+            ),
+        }
+    )
+
+    # Write parquet with arrow for now (to write arrow:schema)
+    buffer = BytesIO()
+    pdf.to_parquet(buffer, engine="pyarrow")
+
+    # Read parquet with arrow schema
+    got = cudf.read_parquet(buffer)
+    # Convert to cudf table for an apple to apple comparison
+    expected = cudf.from_pandas(pdf)
+
+    # Check results for reader with schema
+    assert_eq(expected.dtypes, got.dtypes)
+    assert_eq(expected, got)
+
+    # Now read parquet without arrow schema
+    got = cudf.read_parquet(buffer, use_arrow_schema=False)
+
+    # Check results for reader with schema
+    assert_neq(expected.dtypes, got.dtypes)
+    assert_neq(expected, got)
```

Comment:
Instead, could we construct what should be `expected` by `got` and use `assert_eq`?

Reply:
I think that may lead to some problems in the future if Arrow changes the way they convert from `timedelta` to `int64` during `to_parquet()`. Even now, some columns are being converted as `int64(time in us)` and some as `int64(time in ns)`. 

I can manually construct `expected` and check equality but the tests may become brittle to future changes. What do you think?

Reply:
An alternate solution is to construct an `expected` and check the `dtypes` only as those are deterministic

Reply:
Updated tests to construct a dummy `expected` use `assert_eq` to check the dtypes.